### PR TITLE
Serve actual bundles

### LINKED ISSUE #52: Support real bundle serving
The builder is currently serving empty transactions but we should serve real ones.

### PR DESCRIPTION
**Pull request by QuentinI**
Closes #52

### CODE REVIEW

File: src/service.rs [hunk 1]
```diff
@@ -305,37 +310,148 @@ where
     for<'a> <TYPES::SignatureKey as TryFrom<&'a TaggedBase64>>::Error: Display,
 {
     /// TODO: fetch actual blocks
```
Comment (shenkeyao) [inline src/service.rs:312]: Nit: We can remove this TODO.

File: src/service.rs [hunk 2]
```diff
@@ -305,37 +310,148 @@ where
...
+                .await
+                .spawned_builder_states
+                .iter()
+                .find(|(id, _)| id.view == parent_view)
+                .map(|(id, sender)| (id.clone(), sender.clone()))
+            else {
+                let global_state = self.global_state.read_arc().await;
+                if requested_view <= global_state.last_garbage_collected_view_num
+                    && global_state.highest_view_num_builder_id.view
+                        != global_state.last_garbage_collected_view_num
```
Comment (shenkeyao) [inline src/service.rs:343]: Why do we need this comparison in addition to the first one?

**Review by shenkeyao — COMMENTED** [review]
Minor comments for the current draft. Will re-review when the builder API changes are in.

File: src/service.rs (hunk 2 above)
Comment (QuentinI) [reply src/service.rs]: Added some comments explaining this

**Review by shenkeyao — APPROVED** [review]
Thanks for adding the comments! LGTM.

**Pull request merged**